File: rust_pds/ex2_1.rs
// Rust dont have classes but the structs are much more powerfull
struct Circle{
  radius:f32
}
impl Circle {
  // Rust cannot have optional arguments 
  fn new(radius:f32)->Circle{
    Circle {
      radius
    }
  }
  fn calc_area(&self)->f32{
    std::f32::consts::PI * self.radius * self.radius
  }
  fn calc_circumference(&self)->f32{
    std::f32::consts::PI * 2.0 * self.radius
  }
}
fn main(){
  println!("Radius: ");
  let mut buf = String::new();
  std::io::stdin().read_line(&mut buf).expect("Cannot read from stdin");
  let a:Circle = Circle::new(buf.trim().parse().expect("Invalid number input"));
  println!("Area : {}", a.calc_area());
  println!("Circumference : {}", a.calc_circumference());
}

File: ex2.1.cpp
#include <cmath>
#include <iostream>
using namespace std;
const float DEFAULT_RADIUS = 5.0f;
class Circle {
  float radius;

public:
  Circle(float r = DEFAULT_RADIUS) { radius = r; }
  float calc_area() { return M_PI * radius * radius; }
  float calc_circumference() { return 2 * M_PI * radius; }
};
int main() {
  Circle a;
  cout << "Default Circle\nRadius: " << DEFAULT_RADIUS << endl
       << "Area: " << a.calc_area() << endl
       << "Circumference: " << a.calc_circumference() << endl;
  float radius;
  cout << "Radius: ";
  cin >> radius;
  Circle b(radius);
  cout << "Area: " << b.calc_area() << endl
       << "Circumference: " << b.calc_circumference() << endl;
}
